samples/npm1300_fuel_gauge/src/fuel_gauge.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/npm1300_charger.h>
+#include <zephyr/sys/printk.h>
+
+#include "nrf_fuel_gauge.h"
+
+static float max_charge_current;
+static float max_discharge_current;
+static int64_t ref_time;
+
+static const struct battery_model battery_model = {
+#include "battery_model.inc"
+};
+
+static void read_sensors(const struct device *charger, float *voltage, float *current, float *temp)
+{
+	struct sensor_value value;
+
+	sensor_sample_fetch(charger);
```

Comment:
Check return error codes from `sensor_sample_fetch` and `sensor_channel_get`?
I think it's better to print some error message instead of causing the fuel gauge printouts to become unstable if these values aren't set properly